lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -431,18 +455,50 @@ def create_coefficients_cube(
             coeff_names, long_name="coefficient_name", units="no_unit")
         dim_coords_and_dims = [(coefficient_index, 0)]
         aux_coords_and_dims = [(coefficient_name, 0)]
-        for coord_name in (
-                ["time", "forecast_period", "forecast_reference_time"]):
-            try:
-                aux_coords_and_dims.append(
-                    (current_forecast.coord(coord_name), None))
-            except CoordinateNotFoundError:
-                pass
-        attributes = {"diagnostic_standard_name": current_forecast.name()}
-        for attribute in current_forecast.attributes.keys():
+
+        # Create a forecast_reference_time coordinate.
+        frt_point = cycletime_to_datetime(self.current_cycle)
+        try:
+
+            frt_coord = (
+                historic_forecast.coord("forecast_reference_time").copy(
+                    datetime_to_iris_time(frt_point, time_units="seconds")))
+        except CoordinateNotFoundError:
+            pass
+        else:
+            aux_coords_and_dims.append((frt_coord, None))
+
+        # Create forecast period and time coordinates.
+        try:
+            fp_point, = (
+                np.unique(historic_forecast.coord("forecast_period").points))
+            fp_coord = (
+                historic_forecast.coord("forecast_period").copy(fp_point))
+        except CoordinateNotFoundError:
+            pass
+        else:
+            aux_coords_and_dims.append((fp_coord, None))
+            if historic_forecast.coords("time"):
+                frt_point = cycletime_to_datetime(self.current_cycle)
+                # Ensure that the fp_point is determined with units of seconds.
+                copy_of_fp_coord = (
+                    historic_forecast.coord("forecast_period").copy())
+                copy_of_fp_coord.convert_units("seconds")
+                fp_point, = np.unique(copy_of_fp_coord.points)
```

Comment:
Didn't we just do lot's of this above? Could we do this once rather than twice with a little restructuring? Or is the difference that we no longer want to change to seconds above but we do here?

Reply:
Yes, you're right that there is a bit of duplication here but it is for the purpose of being able to ensure that the forecast period coordinate is in seconds here, so that the construction of the `timedelta` object will work as intended, as this is expecting seconds as input: `datetime.timedelta(seconds=float(fp_point))`.